syntax = "proto3";

import "google/protobuf/timestamp.proto";

package doca_flow_counter_spy;

service CounterSpy {
    rpc getFlowCounts(EmptyRequest) returns (QueryResult) {}
}

message EmptyRequest {}

message Entry
{
    uint64 id = 1;
    uint64 packets = 2;
    uint64 bytes = 3;
}

message Pipe
{
    string name = 1;
    repeated Entry entries = 2;
}

message Port
{
    uint32 port_id = 1;
    repeated Pipe pipes  = 2;
}

message QueryResult
{
    google.protobuf.Timestamp timestamp = 1;
    repeated Port ports = 2;
}
